### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading '0') while I set up my own hedera local network

### ISSUE THREAD

**Issue by CYJ00** [hashgraph/hedera-sdk-js#1502]
### Description

I'm settin up my own local network.

My node is v.14.17.6, npm is v8.5.0, docker is 23.0.1 and docker-compose is 1.29.2.
My @hashgraph/sdk is v2.22.0

This error is ocurred when I run index.js.
![image](https://user-images.githubusercontent.com/126875107/224978998-03b9e9fb-8f71-4c69-b4d2-d57b1609908f.png)

How can I fix it?

### Steps to reproduce

This is my index.js code:
const {
    Client,
    PrivateKey,
    Hbar,
    AccountId,
    AccountCreateTransaction,
} = require("@hashgraph/sdk");

async function main() {
    //Create your local client
    const node = {"127.0.0.1:50211": new AccountId(3)}
    const client = Client.forNetwork(node).setMirrorNetwork("127.0.0.1:5600");
    //Set the transaction fee paying account
    client.setOperator(AccountId.fromString("0.0.2"),PrivateKey.fromString("302e020100300506032b65700..."));
    //Submit a transaction to your local node
    const newAccount = await new AccountCreateTransaction()
        .setKey(PrivateKey.fromString("302e020100300506032b65700..."))
        .setInitialBalance(new Hbar(316))
        .execute(client);
    //Get receipt
    const receipt = await newAccount.getReceipt(client);
    //Get the account ID
    const newAccountId = receipt.accountId;
    console.log(newAccountId);
}
void main();

### Hedera network

other

### Version

v2.22.0

### Operating system

Linux

**Comment by ochikov** [comment]
Hello @CYJ00 and thank you for your bug. The problem is that the Mirror Node of the Local Node is used for updating an address book. This address book is used for fetching Consensus Nodes for the operation of the SDK. This mirror node is returning an address of the consensus node which based on my research is wrong and I've contacted the team which is developing the local node repo. 
TLDR: When you are using the local node, you can remove the `setMirrorNetwork("127.0.0.1:5600")` setter from the client.

**Comment by CYJ00** [comment]
Thank you for your answer!
I did what you said. Then this error is occured.
![image](https://user-images.githubusercontent.com/126875107/225203235-adb3fa79-6ea1-4e63-9725-09ed67af101c.png)

**Comment by ochikov** [comment]
@CYJ00 Please ensure that your local node is up and running. Can you access the local version of Hashscan on http://localhost:8080/devnet/dashboard ?

**Comment by CYJ00** [comment]
![image](https://user-images.githubusercontent.com/126875107/225319731-fb5e2617-3e00-486d-8018-8150f9c0153f.png)

and this is my docker ps
![image](https://user-images.githubusercontent.com/126875107/225319995-a4bbee3d-3390-4864-978a-c08c9d400b11.png)

**Comment by ochikov** [comment]
@CYJ00 It's strange behavior. This is mine as well and it looks like yours. I can see that you've got the service running. I can run your code as well:
![Screenshot 2023-03-17 at 10 35 05](https://user-images.githubusercontent.com/13587218/225854071-a09676b4-3ffc-4285-b37c-7b547c7633a4.png)

![Screenshot 2023-03-17 at 10 33 06](https://user-images.githubusercontent.com/13587218/225854044-c755eb58-8313-4122-8c8c-c58485ca6de1.png)

**Comment by CYJ00** [comment]
<img width="372" alt="스크린샷 2023-03-20 오후 3 18 44" src="https://user-images.githubusercontent.com/126875107/226262465-85d5ec79-5f50-469f-bd19-9c115680b849.png">
I tried same code in Mac, and it worked. I should retry it after removing all node, npm, docker, etc...
Thanks a lot!!

**Comment by ochikov** [comment]
@CYJ00 I will close the issue. The team that is working on the local-node will provide a fix for returning the right Consensus node port when it's run in local environment.
